akkaActorsPersistent/src/main/scala/com/lightbend/scala/modelserver/actor/modelserver/AkkaModelServer.scala

```diff
@@ -46,20 +46,17 @@ object AkkaModelServer {
 
     // Model stream processing
     Consumer.atMostOnceSource(modelConsumerSettings, Subscriptions.topics(MODELS_TOPIC))
-      .map(record => ModelToServe.fromByteArray(record.value())).filter(_.isSuccess).map(_.get)
-      .map(record => ModelWithDescriptor.fromModelToServe(record)).filter(_.isSuccess).map(_.get)
-      .mapAsync(1)(elem => (modelserver ? elem))
-      .to(Sink.ignore) // we do not read the results directly
-      .run() // we run the stream
+      .map(record => ModelToServe.fromByteArray(record.value)).collect { case Success(a) => a }
+      .map(record => ModelWithDescriptor.fromModelToServe(record)).collect { case Success(a) => a }
+      .mapAsync(1)(elem => modelserver ? elem)
+      .runWith(Sink.ignore) // run the stream, we do not read the results directly
```

Comment:
Also guys, In Java8 collect is a very expensive operation. This is one of the reasons I was trying to avoid it. Is it true for Scala as well?

Reply:
Which collect are you refering to in Java? The terminal one? That's completely different.
Collect is filter + map in Scala as well as Akka Streams. It is cheap

Reply:
> Thanks Konrad
> Java 8 Stream collect, I was told

Please stick to using github for review, not private email, thanks.

Well yeah, in Java "streams" (bulk collections) the collect is the "run and accumulate", so totally different and yes, there it is expensive, see docs https://docs.oracle.com/javase/8/docs/api/java/util/stream/Collectors.html